src/freenas/usr/local/lib/middlewared_truenas/plugins/failover.py

```diff
@@ -1419,15 +1442,22 @@ async def journal_ha(middleware):
 
 
 def journal_sync(middleware):
+
+    alert = True
     while True:
         try:
             journal = Journal()
             journal_sync = JournalSync(middleware, sql_queue, journal)
             while True:
                 journal_sync.process()
+        except OSVersionMismatch:
+            if alert:
+                logger.warning('OS version does not match remote node. Not syncing journal')
+                alert = False
         except Exception:
             logger.warning('Failed to sync journal', exc_info=True)
-            time.sleep(5)
```

Comment:
We need to set alert = True again if we had a successful loop pass with matched os version